dom/observable/tentative/observable-ctor.any.js

```diff
@@ -0,0 +1,685 @@
+test(() => {
+  assert_true(
+    typeof Observable === "function",
+    "Observable constructor is defined"
+  );
+}, "Observable constructor");
+
+test(() => {
+  let initializerCalled = false;
+  const results = [];
+
+  const source = new Observable((subscriber) => {
+    initializerCalled = true;
+    subscriber.next(1);
+    subscriber.next(2);
+    subscriber.next(3);
+    subscriber.complete();
+  });
+
+  assert_false(
+    initializerCalled,
+    "initializer should not be called by construction"
+  );
+
+  source.subscribe({
+    next: (x) => results.push(x),
+    error: () => assert_unreached("should not be called"),
+    complete: () => results.push("complete"),
+  });
+
+  assert_true(initializerCalled, "initializer should be called by subscribe");
+  assert_array_equals(
+    results,
+    [1, 2, 3, "complete"],
+    "should emit values synchronously"
+  );
+}, "Observable constructor calls initializer on subscribe");
+
+test(() => {
+  const error = new Error("error");
+  const results = [];
+
+  const source = new Observable((subscriber) => {
+    subscriber.next(1);
+    subscriber.next(2);
+    subscriber.error(error);
+  });
+
+  source.subscribe({
+    next: () => assert_unreached("should not be called"),
+    error: (e) => results.push(e),
+    complete: () => assert_unreached("should not be called"),
+  });
+
+  assert_array_equals(
+    results,
+    [1, 2, error],
+    "should emit error synchronously"
+  );
+}, "Observable error path called synchronously");
+
+test(() => {
+  const error = new Error("error");
+  const results = [];
+
+  const source = new Observable((subscriber) => {
+    try {
+      subscriber.next(1);
+      throw error;
+    } finally {
+      assert_true(subscriber.closed, "subscriber is closed after error");
+    }
+  });
+
+  source.subscribe({
+    next: (x) => results.push(x),
+    error: (e) => results.push(e),
+    complete: () => assert_unreached("should not be called"),
+  });
+
+  assert_array_equals(
+    results,
+    [1, error],
+    "should emit values and the throw error synchronously"
+  );
+}, "Observable should error if initializer throws");
+
+test(() => {
+  let initializerCalled = false;
+  const source = new Observable(() => {
+    initializerCalled = true;
+  });
+
+  assert_false(
+    initializerCalled,
+    "initializer should not be called by construction"
+  );
+  source.subscribe();
+  assert_true(initializerCalled, "initializer should be called by subscribe");
+}, "subscribe can be called with no arguments");
+
+test(() => {
+  const source = new Observable((subscriber) => {
+    subscriber.complete();
+    assert_true(subscriber.closed, "subscriber is closed after complete");
+  });
+
+  source.subscribe();
+}, "subscriber is closed after complete");
```

Comment:
Similar to the above comment about `.closed`. It isn't totally clear to me what this buys us, so maybe we could discuss it on the explainer before writing test expectations around it?

Reply:
`closed` is necessary for "synchronous firehose" scenarios. If you want `take` to work in this scenario:

```ts
const result = new Observable(subscriber => {
  let n = 0;
  while (!subscriber.closed) {
    subscriber.next(n++);
  }
})

result.take(3).subscribe({ 
  next: console.log
})
```

If there's no way to examine if the subscription has been ended by the `take(3)`, then there's no way to break that `while` loop.